src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -375,6 +380,33 @@ public int compareOffsets(final NakadiCursor first, final NakadiCursor second) t
         return KafkaCursor.fromNakadiCursor(first).compareTo(KafkaCursor.fromNakadiCursor(second));
     }
 
+    public long totalEventsInPartition(final Timeline timeline, final String partitionString) {
+        final Timeline.StoragePosition positions = timeline.getLatestPosition();
+        if (positions == null) {
+            throw new InvalidCursorOperation(InvalidCursorOperation.Reason.PARTITION_NOT_FOUND);
+        }
+        final int partition = Integer.valueOf(partitionString);
+        final Timeline.KafkaStoragePosition kafkaPositions = (Timeline.KafkaStoragePosition) positions;
+        final List<Long> offsets = kafkaPositions.getOffsets();
+        if (offsets.size() - 1 < partition) {
+            throw new InvalidCursorOperation(InvalidCursorOperation.Reason.PARTITION_NOT_FOUND);
```

Comment:
In general it seems that this method still "thinks" that it is inside `CursorOperationsService` class :)
It is strange that it throws InvalidCursorOperation because it doesn't have any "cursor" parameter